lib/ASRouter.jsm

```diff
@@ -1823,11 +1864,22 @@ class _ASRouter {
       }));
     } else {
       // On new tab, send cards if they match; othwerise send a snippet
-      message =
-        (await this.handleMessageRequest({
-          provider: "onboarding",
-          template: "extended_triplets",
-        })) || (await this.handleMessageRequest({ provider: "snippets" }));
+      message = await this.handleMessageRequest({
+        provider: "onboarding",
+        template: "extended_triplets",
+      });
+
+      // Set up the experiment for extended triplets. It's done here because we
+      // only want to enroll users (both for control and test) if they meet the
+      // targeting criteria.
+      if (message) {
+        await this.setupExtendedTriplets();
+      }
+
+      if (!(message && this.state.showExtendedTriplets)) {
```

Comment:
even though it's equivalent personally I think `(!message || !this.state.showExtendedTriplets)` is more clear, or maybe put a comment as to what this is checking

Reply:
Also I'd like to see a test for this if you don't mind in the `"#onMessage: NEWTAB_MESSAGE_REQUEST"` block

Reply:
Good point. More tests were added.